packages/algob/src/lib/dryrun.ts

```diff
@@ -266,6 +176,18 @@ export class Tealdbg {
   async run (debugCtxParams?: DebuggerContext): Promise<void> {
     // construct encoded dryrun request using SDK
     const dryRunRequest = await this.createDryRunReq();
+
+    /*
+      Encoding fails on taking empty arrays ([]), so we need to convert
+      to undefined first.
+    */
+    if (dryRunRequest.accounts.length === 0) {
+      (dryRunRequest as any).accounts = undefined;
+    }
+    if (dryRunRequest.apps.length === 0) {
+      (dryRunRequest as any).apps = undefined;
```

Comment:
can you try to not use `as`? We want to limit it, as it already created few problems.

Reply:
[type hack] Need to do it here, as `dryRunReq.apps` has type `modelsv2.Application[]` (but it should be `Application[] | undefined`) here. If you don't set it as `undefined`, `encodeObj` will fail.

I can create an issue on their repo though

Reply:
OK, let's add a comment to explain that

Reply:
ok